wolfcrypt/src/random.c

```diff
@@ -983,6 +983,29 @@ static void Entropy_StopThread(void)
     }
 }
     /* end if defined(HAVE_PTHREAD) */
+
+#elif defined(_WIN32) /* USE_WINDOWS_API */
+
+/* Get the high resolution time counter.
+ *
+ * @return  64-bit time that is the nanoseconds of current time.
+ */
+static WC_INLINE word64 Entropy_TimeHiRes(void)
+{
+    static int           init = 0;
+    static LARGE_INTEGER freq;
+    LARGE_INTEGER        count;
+
+    if (!init) {
+        QueryPerformanceFrequency(&freq);
```

Comment:
Optional: we should check the return from` QueryPerformanceFrequency()` and `QueryPerformanceCounter()`.

This leads me to the question: How do we ensure the function Entropy_TimeHiRes() is performing as anticipated?

Reply:
Good point. There seems to be no way to return / handle an error case. @SparkiDev ?

Reply:
Failure would result in the same value coming out.
We would then see repeating differences which would lead to catastrophic entropy failure.

Reply:
True that it would lead to an entropy failure. Let's hope that the cert lab doesn't require us to implement these changes. This could be flagged as a certification bug under DO-178.